### PR TITLE
test(xsnap): round-trip byte sequences from golang for dynamic IBC

### PR DESCRIPTION
**Pull request by dckc**
refs: #2118

### CODE REVIEW

File: packages/xsnap/test/test-xsnap.js [hunk 1]
```diff
@@ -511,3 +511,28 @@ test('property name space exhaustion: orderly fail-stop', async t => {
...
+  t.teardown(() => vat.terminate());
+
+  await vat.evaluate(`
+    globalThis.send = it => issueCommand(ArrayBuffer.fromString(JSON.stringify(it)));
+  `);
+  function checkStrings() {
+    const octets = new Array(256).fill(0).map((_, i) => i);
+    // We receive some external JSON data (from Golang) in the following format:
+    const bstring1 = String.fromCharCode.apply(null, octets);
+    const bstring2 = JSON.parse(JSON.stringify(bstring1));
```
Comment (michaelfig) [inline packages/xsnap/test/test-xsnap.js:527]: I thought of a further test that may be essential to the situation:

```suggestion
        const jstring1 = JSON.stringify(bstring1);
        const joctets = new Array(256).fill(0).map((_, i) => `\\x${i.toString(16).leftPad(2, '0')}`).join('')
        assert(jstring1 === `"${joctets}"`);
        const bstring2 = JSON.parse(jstring1);
```

Please forgive the untested well-intentions here.  I'm at the end of my day and don't have a clearer way to generate `"\\x00\\x01\\x02...\\xfe\\xff"`.  Perhaps it should be written out in full, for clarity of the test.

**Review by michaelfig — APPROVED** [review]

**Review by kriskowal — APPROVED** [review]
Cool. Round tripping values in the range [0, 256) is a good baseline, especially those outside ASCII.

**Comment by dckc** [comment]
I'm inclined to move on to other things. Let me know if you think it's cost-effective to land this soonish. Or if you want to take it over!

**Comment by michaelfig** [comment]
> I'm inclined to move on to other things. Let me know if you think it's cost-effective to land this soonish. Or if you want to take it over!

It's important for my confidence in IBC, so I will gladly take it on today.

**Pull request closed without merging**